pkg/broker/util.go

```diff
@@ -187,35 +195,44 @@ func createUIFormItem(pd apb.ParameterDescriptor, paramIndex int) (interface{},
 }
 
 // getType transforms an apb parameter type to a JSON Schema type
-func getType(paramType string) schema.PrimitiveTypes {
+func getType(paramType string) (schema.PrimitiveTypes, error) {
 	switch paramType {
 	case "string", "enum":
-		return []schema.PrimitiveType{schema.StringType}
+		return []schema.PrimitiveType{schema.StringType}, nil
 	case "int":
-		return []schema.PrimitiveType{schema.IntegerType}
+		return []schema.PrimitiveType{schema.IntegerType}, nil
 	case "object":
-		return []schema.PrimitiveType{schema.ObjectType}
+		return []schema.PrimitiveType{schema.ObjectType}, nil
 	case "array":
-		return []schema.PrimitiveType{schema.ArrayType}
+		return []schema.PrimitiveType{schema.ArrayType}, nil
 	case "bool", "boolean":
-		return []schema.PrimitiveType{schema.BooleanType}
+		return []schema.PrimitiveType{schema.BooleanType}, nil
 	case "number":
-		return []schema.PrimitiveType{schema.NumberType}
+		return []schema.PrimitiveType{schema.NumberType}, nil
 	case "nil", "null":
-		return []schema.PrimitiveType{schema.NullType}
+		return []schema.PrimitiveType{schema.NullType}, nil
 	}
-	return []schema.PrimitiveType{schema.UnspecifiedType}
+	return nil, fmt.Errorf("Could not find the parameter type for: %v", paramType)
```

Comment:
so what sparked this patch originally?

Reply:
#388 sparked this patch. 

The basics of the issue are that the UnspecifiedType does not marshall into json, therefore the only things I can think to do are:
1. either catch this by converting all `specs` to `services` during load time of the specs 

2. When `/catalog` is called and we convert `specs` to `services` we catch that error/unspecified type and do not add it to the list of `services` but do not blow up the entire response. 

3. blow up the entire response w/ a 500. 


I choose to go down this route because if we can not gracefully handle the Unspecified type in the response then I think we should treat that as an error.